internal/settings/providers/namecom/getrecord.go

```diff
@@ -0,0 +1,67 @@
+package namecom
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/qdm12/ddns-updater/internal/settings/errors"
+	"github.com/qdm12/ddns-updater/internal/settings/utils"
+)
+
+func (p *Provider) getRecordID(ctx context.Context, client *http.Client,
+	recordType string) (recordID int, err error) {
+	u := &url.URL{
+		Scheme: "https",
+		Host:   "api.name.com",
+		Path:   fmt.Sprintf("/v4/domains/%s/records", p.domain),
+		User:   url.UserPassword(p.username, p.password),
+	}
+
+	// by default GET request will return 1000 records.
+	request, err := http.NewRequestWithContext(ctx, http.MethodGet, u.String(), nil)
+	if err != nil {
+		return -1, fmt.Errorf("%w: %w", errors.ErrBadRequest, err)
```

Comment:
nit you can return `0`, anyway the caller should discard it if an error is returned. `-1` looks like a special number the caller may have to handle in case of errors 🤔